internal/store/data_lake.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongocli/internal/config"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+//go:generate mockgen -destination=../mocks/mock_data_lake.go -package=mocks github.com/mongodb/mongocli/internal/store DataLakeLister,DataLakeDescriber,DataLakeCreator,DataLakeDeleter,DataLakeUpdater,DataLakeStore
+
+type DataLakeLister interface {
```

Comment:
[aside] We try to add them as we need them to avoid cases where "you ain't gonna need it" but let's not remove them since they're done

Reply:
yeah this was mostly prepping as I knew they were all going to be needed but it made this pr longer. I'll fix the comment typo in the next PR

Reply:
Actually I probably need a new commit to pick up the library anyway, so I can do it now